File: Arrays/ChocolateDistribution.py
#User function Template for python3

class Solution:

    def findMinDiff(self, A,N,M):
         
        mini=max(A)+1
        A.sort()
        for i in range(0,N-M+1):
            if(A[i+M-1]-A[i]<mini):
                mini=A[i+M-1]-A[i]
        
        return mini

        # code here

#{ 
#  Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':

    t = int(input())

    for _ in range(t):
        N = int(input())
        A = [int(x) for x in input().split()]
        M = int(input())


        solObj = Solution()

        print(solObj.findMinDiff(A,N,M))
# } Driver Code Ends

File: Arrays/ChocolateDistribution.java
// { Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
class GfG
{
    public static void main (String[] args)
    {
        
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            long n = sc.nextLong();
            ArrayList<Long> arr = new ArrayList<Long>();
            for(int i = 0;i<n;i++)
                {
                    long x = sc.nextInt();
                    arr.add(x);
                }
            long m = sc.nextLong();
            
            Solution ob = new Solution();
            
    		System.out.println(ob.findMinDiff(arr,n,m));
        }
        
    }
}// } Driver Code Ends


//User function Template for Java

class Solution
{
    public long findMinDiff (ArrayList<Long> a, long n, long m)
    {
        // your code here
         Long mini=Long.MAX_VALUE;;
         int nn=(int)n;
         int mm=(int)m;
        Collections.sort(a);
        for(int i=nn-1;i-mm+1>=0;i--){
            Long max=a.get(i);
           // System.out.println(max);
            Long min=a.get(i-mm+1);
            //System.out.println(min);
            
            //System.out.println();
        
            if(max-min<mini)
            mini=max-min;
        }
      return mini;
        
    }
    
}
